network-api/networkapi/buyersguide/views.py

```diff
@@ -1,23 +1,84 @@
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+from django.db import Error
 from django.shortcuts import render
-from networkapi.buyersguide.models import Product
 from django.contrib.auth.decorators import login_required
 from django.conf import settings
+from django.views.decorators.csrf import csrf_protect
+from rest_framework.parsers import JSONParser
+from rest_framework.decorators import api_view, parser_classes, throttle_classes, permission_classes
+from rest_framework.response import Response
+from rest_framework.permissions import AllowAny
+
+from networkapi.buyersguide.models import Product, BooleanVote, RangeVote
+from networkapi.buyersguide.throttle import UserVoteRateThrottle, TestUserVoteRateThrottle
+from networkapi.buyersguide.serializers import ProductSerializer
+
+vote_throttle_class = UserVoteRateThrottle if not settings.TESTING else TestUserVoteRateThrottle
 
 
 # Login required so we can continue to develop this and merge into master without the public seeing it.
 # Remove this when we're ready to launch.
 @login_required
 def buyersguide_home(request):
     products = Product.objects.all()
-    return render(request, 'buyersguide_home.html', {'products': products})
+    serialized_products = [ProductSerializer(product) for product in products]
+    return render(request, 'buyersguide_home.html', {'products': serialized_products})
```

Comment:
In this case the dict being passed here is the context when rendering the template. (I think)

Reply:
is there a reason we're serializing it then? As opposed to just passing in the model instances directly?

Reply:
We're using the serializer to generate a `votes` attribute that presents the aggregate vote data in a nice format.

Reply:
serializer seems to be extra overhead for just that one thing though. Are you able to just do it in this function itself since you need the rest of the model data anyway?

Reply:
Maybe, I'll look into an alternative implementation